git.go

```diff
@@ -24,7 +24,7 @@ const (
 var sshExp = regexp.MustCompile("^(?P<sshUser>[^@]+)@(?P<domain>[^:]+):(?P<pathRepo>.*)\\.git/?$")
 
 // https://github.com/Shyp/shyp_api.git
-var httpsExp = regexp.MustCompile("^https://(?P<domain>[^/:]+)(:(?P<port>[[0-9]+))?/(?P<pathRepo>.*)\\.git/?$")
+var httpsExp = regexp.MustCompile("^https://(?P<domain>[^/:]+)(:(?P<port>[[0-9]+))?/(?P<pathRepo>.+?)(\\.git/?)?$")
```

Comment:
Changing the asterisk to a question mark says "be lazy, not greedy" so we don't try to snatch a `.git` as part of the `pathRepo`.

The addition of the parentheses here `(\\.git/?)?` makes the `.git` part of the string optional.